pyanaconda/payload/rpmostreepayload.py

```diff
@@ -195,32 +152,17 @@ def _install(self, data):
         from gi.repository import OSTree, RpmOstree
         log.info("executing ostreesetup=%r", data)
 
-        # Initialize the filesystem - this will create the repo as well
-        self._safe_exec_with_redirect("ostree",
-                                      ["admin", "--sysroot=" + conf.target.physical_root,
-                                       "init-fs", conf.target.physical_root])
+        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import \
+            InitOSTreeFsAndRepoTask
+        task = InitOSTreeFsAndRepoTask()
```

Comment:
I am not sure if we need to be so granular with the `_install` method. I would move it all into one task. Mainly because I don't understand it either :-)

Reply:
We'll see if my attempt survives actual testing. Such an approach would be definitely better if things break too much.